src/index.js

```diff
@@ -842,6 +935,82 @@ export default function ({ types: t, template }) {
         catch (error) {
           buildCodeFrameError(path, error)
         }
+      },
+
+      VariableDeclaration(path, state) {
+        if (state.opts[SKIP_ASSERTS_OPTION]) {
+          return
+        }
+
+        const node = path.node
+        try {
+          node.declarations.forEach(declarator => {
+            const id = declarator.id
+
+            if (id.hasBeenTypeChecked) {
+              return declarator
+            }
+
+            const typeAnnotation = id.typeAnnotation && id.typeAnnotation.typeAnnotation
+            id.savedTypeAnnotation = typeAnnotation
+
+            if (!declarator.init || !id.savedTypeAnnotation) {
+              return
+            }
+
+            declarator.init = getWrappedVariableDeclaratorInitWithTypeCheckAST(declarator)
+            id.hasBeenTypeChecked = true
+          })
```

Comment:
here and in `AssignmentExpression` a `hasAsserts = true` is missing (otherwise the `_assert` helper wont be inserted)